Source/BoundaryConditions/WarpX_PEC.cpp

```diff
@@ -49,22 +49,32 @@ PEC::ApplyPECtoEfield (std::array<std::unique_ptr<amrex::MultiFab>, 3>& Efield,
     amrex::IntVect Ex_nodal = Efield[0]->ixType().toIntVect();
     amrex::IntVect Ey_nodal = Efield[1]->ixType().toIntVect();
     amrex::IntVect Ez_nodal = Efield[2]->ixType().toIntVect();
-    int nComp_x = Efield[0]->nComp();
-    int nComp_y = Efield[1]->nComp();
-    int nComp_z = Efield[2]->nComp();
+    // For each Efield multifab, apply PEC boundary condition to ncomponents
+    // If not split E-field, the PEC is applied to the regular Efield used in Maxwell's eq.
+    // If split_pml_field is true, then PEC is applied to the first two components of the pml multifabs
+    // Since PML_components are xy, xz for Efield[0]
+    //                          yz, yx for Efield[1]
+    //                          zx, zy for Efield[2]
+    int nComp_x = (split_pml_field) ? 2 : Efield[0]->nComp();
```

Comment:
@RemiLehe  @dpgrote Please let me know if you have a suggestion here to avoid using 2 as a hard-coded integer and use ncomp from split fields. Unless divE cleaning is on, pml multifabs do not have 3 components, they always have 2. But I wanted to keep this general, so left this as 2 here.

Reply:
Good point! I'm thinking, why don't we simply call the method `nComp()` regardless of whether it's called from a MultiFab in the valid domain or from the corresponding MultiFab in the PML (so simply `int nComp_x = Efield[0]->nComp();` and so on, as it was before)? I think such a call would return `1` component for the MultiFabs in the valid domain and `2` or `3` components, automatically, for the MultiFabs in the PML (`2` if div(E) cleaning is OFF, `3` if div(E) cleaning is ON). Do you think this would work or am I overlooking something?

Another separate question (for @RevathiJambunathan, @RemiLehe and @dpgrote): Do we need to apply these boundary conditions to the F MultiFab as well, when div(E) cleaning is ON. Both in the valid domain (where F is a scalar) and in the PML (where F has three components Fx, Fy, Fz)?

Reply:
See comment above. The reason I asked is because, when nComp() is 3 (when divE cleaning is on), the 3rd component is not tangential to the surface, but normal . Atleast that is what I understand from `xx` defined in PMLComponent.H. And PEC should only set the tangential components of the E-field to 0.

Reply:
Thank you for the clarifications, Reva! I added a point on the agenda for tomorrow's meeting to discuss what to do with F (and, if necessary, what to do with the diagonal components of E, such `xx` etc.).

Reply:
great! I was just about to -- you beat me to it!!!! :)